centrallix-doc/Report/report.xml

```diff
@@ -343,14 +344,20 @@ myTable "report/table"
 		<properties>
 			<property name="box" type="yes/no">Whether to draw a box around the chart's data area (the box is drawn just around the graph or chart itself, not around the axis and tick labels and titles).  Default "no".</property>
 			<property name="chart_type" type="string">The type of chart:  bar, line, or pie.</property>
-			<property name="color" type="string">The default color to be used for the chart.</property>
+			<property name="color" type="string">The default color to be used for the chart. If nothing is specified elsewhere, all series will be made this color. If a palette is also specified, this color will replace any invalid colors.</property>
+			<property name="palette" type="string">A comma separated list of colors to be used by charts with dynamic series. This takes priority over the color property and the default palette. Once the palette runs out of to map to series, it loops through the colors again. Example: "green,,#014421", would have series colored a bright green, the default, and then a dark green.</property>
```

Comment:
For the `palette` property, I'd recommend doing a structure-file-parsed comma separated list instead of a comma separated list inside a string.  That way, individual elements of the list are easier to specify with an expression or whatnot.  For example:

`palette = "green", "red", "blue";`

instead of

`palette = "green, red, blue";`

The updated list can be retrieved just using rpt_internal_GetString() calls with successively higher index (nval) values.  Just use a NULL default value and stop iterating when a NULL comes back.

Reply:
Continuing to noodle on this -- looking ahead, if we want to have a system-wide chart palette preference, we'd need a way to SELECT a list of colors from a preference table.  I'd just need to make sure the expression language and report writer could handle a list created by a subquery expression.  On the other hand, in the short term, it might actually be useful for the report writer code to handle either of the above cases - parsing colors in a list from a single string vs. having them in separate strings.

Reply:
Okay! The driver can now handle any mixture of lists of strings and strings containing lists.